dao/mongo.go

```diff
@@ -105,41 +104,54 @@ func (m *MongoService) CreateInsolvencyResource(dao *models.InsolvencyResourceDt
 	return http.StatusConflict, fmt.Errorf("an insolvency case already exists for this transaction id")
 }
 
-// GetInsolvencyResource retrieves all the data for an insolvency case with the specified transactionID
-func (m *MongoService) GetInsolvencyResource(transactionID string) (models.InsolvencyResourceDao, error) {
+// GetInsolvencyPractitionersResource retrieves all the data for an insolvency case with the specified transactionID
+func (m *MongoService) GetInsolvencyPractitionersResource(transactionID string) (*models.InsolvencyResourceDao, []models.PractitionerResourceDao, error) {
```

Comment:
This feels like two pieces of functionality to me (retrieval of insolvency + practitioner details), especially as they are being returned as 2 separate objects and the insolvency is retrieved without the practitioners several times (e.g. in soa_service & similar services).  Should we consider either splitting them (e.g. by bringing getInsolvencyPractitionersDetails up into the interface) or revisiting how the practitioners are retrieved (e.g. by adding the transactionID to the practitioners collection)? (This might also remove the need to pass in the additional argument to e.g. ValidateInsolvencyDetails in insolvency_service)

Reply:
I agree that having multiple return values is a cool capability of Go, but the examples of best practice I've seen tend to refer to error handling etc. or another closely related output of the function's core purpose, whereas this still feels to me like 2 separate tasks producing 2 separate results: everything before line 133 retrieves the insolvency resource, & everything after retrieves the practitioners.  I would still be in favour of splitting this into 2 functions with clearly defined single responsibilities.